File: src/main/java/org/jonnyzzz/demo/ProcessStrings.java
package org.jonnyzzz.demo;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

public class ProcessStrings {
    private static final String prefix = "prefix";

    public static List<String> main1(List<String> args) {
        final List<String> result = new ArrayList<>();
        for (String arg : args) {
            if (arg.startsWith(prefix)) {
                result.add(arg.substring(prefix.length()));
            }
        }

        return result;
    }

    public static List<String> main2(List<String> args) {

        return
                args.stream()
                .filter(it -> it.startsWith(prefix))
                .map(it -> it.substring(prefix.length()))
                .collect(Collectors.toList());

    }

    public static List<String> main3(List<String> args) {

        return
                args.stream()
                        .map(it -> {
                            if (it.startsWith(prefix)) {
                                return it.substring(prefix.length());
                            } else {
                                return null;
                            }
                        })
                        .filter(Objects::nonNull)
                        .collect(Collectors.toList());

    }
}

File: src/main/java/org/jonnyzzz/demo/ProcessStrings.kt
@file:Suppress("PackageDirectoryMismatch")

package org.jonnyzzz.demo.processStrings

import java.util.stream.Collectors


private const val prefix = "prefix"


private fun main2(args: List<String>): List<String>? {

    return (

            args.stream()
                .filter { it.startsWith(prefix) }
                .map { it.substring(prefix.length) }
                .collect(Collectors.toList())


            )
}

private fun main3(args: List<String>): List<String> {

    return (

            args
                .filter { it.startsWith(prefix) }
                .map { it.substring(prefix.length) }


            )
}


fun main4(args: List<String>): List<String> {
            return (

            args.mapNotNull {
                if (it.startsWith(prefix)) {
                    it.substring(prefix.length)
                } else {
                    null
                }
            }

            )
}


fun main5(args: List<String>): List<String> {

    fun String.removePrefixOrNull(prefix: String) : String? = when {
        startsWith(prefix) -> substring(prefix.length)
        else -> null
    }


    return args.mapNotNull { it.removePrefixOrNull(prefix) }
}

fun main6(args: List<String>): List<String> {

    fun String.removePrefixOrNull(prefix: String) : String? = when {
        startsWith(prefix) -> substring(prefix.length)
        else -> null
    }

    return args.mapNotNull { it.removePrefixOrNull(prefix) }
}
